salt/iiif/config/etc-nginx-sites-enabled-loris-container.conf

```diff
@@ -59,41 +56,15 @@ server {
         uwsgi_pass loris;
         include uwsgi_params;
 
+        # cache result 'forever' (a year, in seconds)
         add_header Cache-Control "public, max-age=31536000, immutable";
-        # http://agentzh.blogspot.co.uk/2011/03/how-nginx-location-if-works.html
-        # see also: https://nginx.org/en/docs/http/ngx_http_core_module.html#var_arg_
-        if ($arg_fallback) {
-            add_header Cache-Control "public, max-age=31536000, immutable";
-            add_header X-Iiif-Fallback 1;
-        }
-
-        uwsgi_intercept_errors on;
-        error_page 500 502 = @iiif_fallback;
     }
 
     location ~ .*/info.json$ {
         uwsgi_pass loris;
         include uwsgi_params;
 
+        # cache result for five minutes
         add_header Cache-Control "public, max-age=300";
-        if ($arg_fallback) { 
```

Comment:
trying to verify that this is not hit anymore in production. Will be back.

Reply:
I see these types of errors in the docker logs:

```
2020-03-27 09:10:43,013 (loris.resolver) [INFO]: Copied https://s3-external-1.amazonaws.com/prod-elife-published/articles/47776/elife-47776-fig4-figsupp3-v3.tif to /usr/local/share/images/loris/lax/81/8e8/da4/7e9/ee6/3d5/73d/ed3/e07/84c/40d/lax:47776/elife-47776-fig4-figsupp3-v3.tif/loris_cache.tif
2020-03-27 09:10:43,150 (loris.resolver) [INFO]: Copied https://s3-external-1.amazonaws.com/prod-elife-published/articles/50033/elife-50033-fig6-figsupp2-v2.tif to /usr/local/share/images/loris/lax/a8/a93/96e/ec9/42c/f9b/5e1/415/a05/0c9/161/lax:50033/elife-50033-fig6-figsupp2-v2.tif/loris_cache.tif
2020-03-27 09:10:43,171 (loris.resolver) [INFO]: Copied https://s3-external-1.amazonaws.com/prod-elife-published/articles/46935/elife-46935-fig5-figsupp1-v1.tif to /usr/local/share/images/loris/lax/4a/25d/819/54f/945/1f1/0cb/664/ed0/9f2/c5a/lax:46935/elife-46935-fig5-figsupp1-v1.tif/loris_cache.tif
2020-03-27 09:10:43,180 (loris.resolver) [INFO]: Copied https://s3-external-1.amazonaws.com/prod-elife-published/articles/50033/elife-50033-fig4-figsupp1-v2.tif to /usr/local/share/images/loris/lax/fa/f72/38a/2e0/7c7/ccb/52a/88e/56d/bec/4c8/lax:50033/elife-50033-fig4-figsupp1-v2.tif/loris_cache.tif
[pid: 13|app: 0|req: 4862/10865] 10.0.3.69 () {34 vars in 396 bytes} [Fri Mar 27 09:10:43 2020] GET / => generated 220 bytes in 2 msecs (HTTP/1.1 200) 5 headers in 166 bytes (3 switches on core 0)
2020-03-27 09:10:43,265 (loris.resolver) [INFO]: Copied https://s3-external-1.amazonaws.com/prod-elife-published/articles/46935/elife-46935-fig9-figsupp1-v1.tif to /usr/local/share/images/loris/lax/e6/a1f/d35/26d/c7d/bf2/e91/970/9e6/999/92e/lax:46935/elife-46935-fig9-figsupp1-v1.tif/loris_cache.tif
[pid: 12|app: 0|req: 1327/10866] 157.52.79.64 () {76 vars in 1756 bytes} [Fri Mar 27 09:10:43 2020] GET /lax:50033%2Felife-50033-fig4-figsupp1-v2.tif/full/1234,/0/default.jpg => generated 93634 bytes in 324 msecs (HTTP/1.1 200) 8 headers in 510 bytes (365 switches on core 0)
2020-03-27 09:10:43,394 (loris.resolver) [INFO]: Copied https://s3-external-1.amazonaws.com/prod-elife-published/articles/46935/elife-46935-fig1-figsupp1-v1.tif to /usr/local/share/images/loris/lax/b3/3d6/875/b19/0f8/21e/575/a0d/cc9/4fb/896/lax:46935/elife-46935-fig1-figsupp1-v1.tif/loris_cache.tif

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/newrelic/api/wsgi_application.py", line 665, in _nr_wsgi_application_wrapper_
    environ, _start_response, transaction)
  File "/usr/local/lib/python3.6/dist-packages/newrelic/api/wsgi_application.py", line 193, in __init__
    self.start_response)
  File "/opt/loris/loris/webapp.py", line 481, in __call__
    return self.wsgi_app(environ, start_response)
  File "/opt/loris/loris/webapp.py", line 424, in wsgi_app
    response = self.route(request)
  File "/opt/loris/loris/webapp.py", line 475, in route
    return self.get_img(request, ident, region, size, rotation, quality, fmt, base_uri)
  File "/opt/loris/loris/webapp.py", line 619, in get_img
    info = self._get_info(ident, request, base_uri)[0]
  File "/opt/loris/loris/webapp.py", line 559, in _get_info
    info = self.resolver.resolve(self, ident, base_uri)
  File "/opt/loris/loris/resolver.py", line 378, in resolve
    return ImageInfo(app=app, src_img_fp=cached_file_path, src_format=format_, auth_rules=auth_rules)
  File "/opt/loris/loris/img_info.py", line 105, in __init__
    self.from_image_file(formats, app.max_size_above_full)
  File "/opt/loris/loris/img_info.py", line 177, in from_image_file
    self._extract_with_pillow(self.src_img_fp)
  File "/opt/loris/loris/img_info.py", line 188, in _extract_with_pillow
    im = Image.open(fp)
  File "/usr/local/lib/python3.6/dist-packages/PIL/Image.py", line 2818, in open
    raise IOError("cannot identify image file %r" % (filename if filename else fp))
OSError: cannot identify image file '/usr/local/share/images/loris/lax/b3/3d6/875/b19/0f8/21e/575/a0d/cc9/4fb/896/lax:46935/elife-46935-fig1-figsupp1-v1.tif/loris_cache.tif'
[pid: 9|app: 0|req: 521/10867] 157.52.79.53 () {74 vars in 1606 bytes} [Fri Mar 27 09:10:43 2020] GET /lax:46935%2Felife-46935-fig1-figsupp1-v1.tif/full/617,/0/default.webp => generated 0 bytes in 393 msecs (HTTP/1.1 500) 0 headers in 0 bytes (0 switches on core 0)
2020-03-27 09:10:43,491 (loris.resolver) [INFO]: Another process downloaded src image /usr/local/share/images/loris/lax/b3/3d6/875/b19/0f8/21e/575/a0d/cc9/4fb/896/lax:46935/elife-46935-fig1-figsupp1-v1.tif/loris_cache.tif
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/newrelic/api/wsgi_application.py", line 665, in _nr_wsgi_application_wrapper_
    environ, _start_response, transaction)
  File "/usr/local/lib/python3.6/dist-packages/newrelic/api/wsgi_application.py", line 193, in __init__
    self.start_response)
  File "/opt/loris/loris/webapp.py", line 481, in __call__
    return self.wsgi_app(environ, start_response)
  File "/opt/loris/loris/webapp.py", line 424, in wsgi_app
    response = self.route(request)
  File "/opt/loris/loris/webapp.py", line 475, in route
    return self.get_img(request, ident, region, size, rotation, quality, fmt, base_uri)
  File "/opt/loris/loris/webapp.py", line 619, in get_img
    info = self._get_info(ident, request, base_uri)[0]
  File "/opt/loris/loris/webapp.py", line 559, in _get_info
    info = self.resolver.resolve(self, ident, base_uri)
  File "/opt/loris/loris/resolver.py", line 378, in resolve
    return ImageInfo(app=app, src_img_fp=cached_file_path, src_format=format_, auth_rules=auth_rules)
  File "/opt/loris/loris/img_info.py", line 105, in __init__
    self.from_image_file(formats, app.max_size_above_full)
  File "/opt/loris/loris/img_info.py", line 177, in from_image_file
    self._extract_with_pillow(self.src_img_fp)
  File "/opt/loris/loris/img_info.py", line 188, in _extract_with_pillow
    im = Image.open(fp)
  File "/usr/local/lib/python3.6/dist-packages/PIL/Image.py", line 2818, in open
    raise IOError("cannot identify image file %r" % (filename if filename else fp))
OSError: cannot identify image file '/usr/local/share/images/loris/lax/b3/3d6/875/b19/0f8/21e/575/a0d/cc9/4fb/896/lax:46935/elife-46935-fig1-figsupp1-v1.tif/loris_cache.tif'
```

is it a cache miss do you suppose? Its preceeded by a bunch of copy operations from S3. I can successfully request the file and get a binary output after the fact.

Reply:
```
curl -o /dev/null -v localhost/lax:49212/elife-49212-fig3-figsupp5-v2.tif/full/full/0/default.jpg
```
on a IIIF server reproduces this error, but works with 200 OK as it falls back to the JPG source. The output headers confirm:
```
X-Iiif-Fallback: 1
```

A quick count of these errors on a single server, with a container that has been active for 4 days:
```
elife@prod--iiif:~$ docker logs loris_uwsgi_1 2>&1 | grep "cannot identify" | wc -l
5016
```
So this fallback has to stay in place it seems.

Reply:
reverted changes